src/Files.App/Data/Models/ItemViewModel.cs

```diff
@@ -970,18 +969,23 @@ private async Task LoadItemThumbnailAsync(ListedItem item, uint thumbnailSize =
 				{
 					await dispatcherQueue.EnqueueOrInvokeAsync(async () =>
 					{
+						// Assign the thumbnail/icon to the listed item
 						item.FileImage = await iconInfo.IconData.ToBitmapAsync();
-						if (!string.IsNullOrEmpty(item.FileExtension) &&
-							!item.IsShortcut && !item.IsExecutable &&
-							!ImagePreviewViewModel.ContainsExtension(item.FileExtension.ToLowerInvariant()))
+
+						// Add the file icon to the DefaultIcons list
+						if (!DefaultIcons.ContainsKey(item.FileExtension.ToLowerInvariant()) && !string.IsNullOrEmpty(item.FileExtension))
 						{
-							DefaultIcons.AddIfNotPresent(item.FileExtension.ToLowerInvariant(), item.FileImage);
+							var fileIcon = await FileThumbnailHelper.LoadIconAndOverlayAsync(item.ItemPath, thumbnailSize, false, true);
+							var bitmapImage = await fileIcon.IconData.ToBitmapAsync();
+							DefaultIcons.Add(item.FileExtension.ToLowerInvariant(), bitmapImage);
```

Comment:
```suggestion
							DefaultIcons.TryAdd(item.FileExtension.ToLowerInvariant(), bitmapImage);
```
It should be considered that other threads may add first between `ContainsKey` and `Add`.